astropy/coordinates/representation.py

```diff
@@ -348,11 +361,14 @@ def __repr__(self):
         prefixstr = '    '
         arrstr = _array2string(self._values, prefix=prefixstr)
 
+        diffstr = ''
+        if getattr(self, 'differentials', None):
```

Comment:
It wasn't immediately clear to me how we would want `__str__` to look when there are differentials. Also, to me, the repr seems like a more appropriate place to expose more of the internals of the object, so I would actually vote to have them appear in the `__repr__` and not in str, but am open to suggestions.